content/6_Advanced/SCC.mdx

```diff
@@ -202,11 +202,92 @@ int main() {
 
 <FocusProblem problem="satEx" />
 
-<IncompleteSection>
+Model each topping as a boolean variable and each wish as a clause, so all wishes define a boolean formula in 2-CNF. Then we rewrite each clause $x_i \vee x_j$ as $(\bar x_i \rarr x_j) \wedge (\bar x_j \rarr x_i)$ and construct the implication graph with each variable and its negation as a vertex. For each appearance of $u \rarr v$ in the rewritten formula, an edge from $u$ to $v$ is inserted into the implication graph. (For a diagram, see the explanation at cp-algo linked below.) Then, the formula is satisfiable if and only if no variable lies in the same Strongly Connected Component as its negation, since this would form a contradiction. So we find the SCC of each variable using Kosarajus algorithm. If the formula is satisfiable, we need to pick the value for a variable in the topologically greater component, as otherwise a contradiction could be formed.
```

Comment:
```suggestion
Model each topping as a boolean variable and each wish as a clause, so all wishes define a boolean formula in 2-CNF. Then we rewrite each clause $x_i \vee x_j$ as $(\bar x_i \rarr x_j) \wedge (\bar x_j \rarr x_i)$ and construct the implication graph with each variable and its negation as a vertex. For each appearance of $u \rarr v$ in the rewritten formula, an edge from $u$ to $v$ is inserted into the implication graph. (For a diagram, see the explanation at cp-algo linked below.) Then, the formula is satisfiable if and only if no variable lies in the same Strongly Connected Component as its negation, since this would form a contradiction. So we find the SCC of each variable using Kosaraju's algorithm. If the formula is satisfiable, we need to pick the value for a variable in the topologically greater component, as otherwise a contradiction could be formed.
```

Reply:
Oh, sorry, I thought you dont use apostrophes in general since you changed it at "its" :)